packages/snaps-execution-environments/src/common/test-utils/testUtils.ts

```diff
@@ -0,0 +1,350 @@
+ *
+ * @param subject - Test subject (instance, object, function).
+ * @param factory - Factory that creates an instance using constructor function.
+ * @returns Array of error messages.
+ */
+export function endowmentHardeningTestFunction(
+  subject: HardenedEndowmentSubject,
+  factory: () => HardenedEndowmentInstance,
+): unknown[] {
+  const log = [];
+  const instance = factory();
+  try {
+    subject.__flag = 'not_secure';
+  } catch (error) {
+    log.push(error.message);
+  }
+  try {
+    if (instance) {
+      instance.__flag = 'not_secure';
+    }
+  } catch (error) {
+    log.push(error.message);
+  }
+  try {
+    subject.prototype.__flag = 'not_secure';
+  } catch (error) {
+    log.push(error.message);
+  }
+  try {
+    // eslint-disable-next-line no-proto
+    instance.__proto__.__flag = 'not_secure';
+  } catch (error) {
+    log.push(error.message);
+  }
+  try {
+    // @ts-expect-error Test unusual approach for a security reasons.
+    // eslint-disable-next-line no-proto
+    subject.__proto__.__flag = 'not_secure';
+  } catch (error) {
+    log.push(error.message);
+  }
+  return log;
+}
+
+/**
+ * Object walker test utility function.
+ * This function will instantiate and configure @lavamoat/lavatube for testing
+ * endowment specific use case. It will also adapt its result to a boolean value.
+ *
+ * @param subject - Subject to be tested.
+ * @param target - Target object.
+ * @returns True if target object is found, false otherwise.
+ */
+export function walkAndSearch(subject: unknown, target: unknown) {
+  let result = false;
+  const walker = new LavaTube(
+    (value: unknown) => {
+      result = target === value;
+      return result;
+    },
+    { maxRecursionLimit: 100, onShouldIgnoreError: () => true },
+  );
+  walker.walk(subject);
+  return result;
+}
+
+/**
+ * Create and return mocked stream provider instance used for testing.
+ * Note: Stream is wrapped in a Proxy to simulate limited provider used in
+ * BaseSnapExecutor for ethereum endowment.
+ *
+ * @returns Proxy to StreamProvider instance.
+ */
+export function getMockedStreamProvider() {
+  const mux = new ObjectMultiplex();
+  const rpcStream = mux.createStream(SNAP_STREAM_NAMES.JSON_RPC);
+
+  const provider = new StreamProvider(rpcStream, {
+    jsonRpcStreamName: 'metamask-provider',
+    rpcMiddleware: [createIdRemapMiddleware()],
+  });
+
+  const originalRequest = provider.request.bind(provider);
+
+  const request = async (args: RequestArguments) => {
+    assert(
+      !args.method.startsWith('snap_'),
+      ethErrors.rpc.methodNotFound({
+        data: {
+          method: args.method,
+        },
+      }),
+    );
+    return await withTeardown(originalRequest(args), { lastTeardown: 0 });
+  };
+
+  // To harden and limit access to internals, we use a proxy.
+  // Proxy target is intentionally set to be an empty object, to ensure
+  // that access to the prototype chain is not possible.
+  return new Proxy(
```

Comment:
This is the same as the other proxy, so should be using the same utility function as well.

Reply:
Extracted to utility function and applied/refactored.